addons/binding/org.openhab.binding.enocean/ESH-INF/thing/ContactAndSwitch.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="enocean"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="contactSwitch">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="serialbridge" />
+			<bridge-type-ref id="tcpbridge" />
+		</supported-bridge-type-refs>
+
+		<label>D5-00 - Single input contact</label>
+		<description>Single input contact which sends EEP D5-00-01</description>
+
```

Comment:
Why are there no channels defined for this thing type? Same question for the others.

Reply:
A thing type can handle EEPs of a certain EEP family. However the EEPs (although belonging to the same family) provide different functionality. For example the EEP family RoomOperatingPanel A5-10 has profiles which can set a fan stage, set an occupancy switch or send temperature or humidity sensor data. As I did not want to create different thing types of all of these combinations, I create the channels dynamically upon the current choosen EEP. In this way I only need one thing type => RoomOperatingPanel.

Reply:
ok, thanks.